components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityClaimValueEncryptionListener.java

```diff
@@ -47,6 +50,11 @@ public class IdentityClaimValueEncryptionListener extends AbstractIdentityUserOp
 
     private static final String CLAIM_VALUE = "ClaimValue";
     private static final String CLAIM_URI = "claimURI";
+    private static final String TOTP_KEY = "CryptoService.TotpSecret";
+    // List of claims related to TOTP.
+    private static final List<String> CLAIMS_FOR_TOTP =
```

Comment:
why do we have totp specific code in the framework? can't this be added in TOTP level?

Reply:
When these claims are updated, all the claims are getting encrypted in this listener based on the `enableEncrption` attribute for a claim. Therefore all the claims are encrypted in this flow.